### ISSUE TITLE
Legend order inconsistency between Bar and Marimekko chart

### ISSUE THREAD

**Issue by archinul** [plouc/nivo#2261]
First of all, thanks a lot for this very cool library.

**Describe/explain the bug**
I'm not an expert so far, but I was playing with the examples on the Website and it seems that the default order for both Bar (stacked) and Marimekko chart legends are different when choosing the column direction.

Intuitively, I would say that the right legend order is the Bar chart one, thats follows the same order as the bars categories.

**To Reproduce**

1. Go to https://nivo.rocks/bar/
2. Set `legends.0.direction` as `column`
3. Go to https://nivo.rocks/marimekko/
4. Set `legends.0.direction` as `column`

**Expected behavior**
The legend order should be the same as the bar categories.

**Screenshots**
<img width="955" alt="image" src="https://user-images.githubusercontent.com/32000733/224354993-23097f08-eab4-45f2-b7fa-9dffa4ddb36a.png">

<img width="946" alt="image" src="https://user-images.githubusercontent.com/32000733/224353588-2354b2bb-2b67-4da5-9dab-6c76ab0b469f.png">

**Additional context**
I spent some time playing with the Marimekko component to see if there's a way to use custom labels by defining the `legends[0].data` but it doesn't seem to have any effect.

**Comment by plouc** [comment]
I agree, the marimekko chart should behave the same way, iirc a fix was submitted for the bar chart in the past, but the marimekko chart wasn't considered.

**Comment by kleinaber** [comment]
same goes for the line chart. :-)
